src/compiler/scanner.ts

```diff
@@ -2684,26 +2688,32 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
                                     error(Diagnostics.Incomplete_quantifier_Digit_expected, digitsStart, 0);
                                 }
                                 else {
-                                    if (unicodeMode) {
-                                        error(Diagnostics.Unexpected_0_Did_you_mean_to_escape_it_with_backslash, start, 1, String.fromCharCode(ch));
-                                    }
+                                    error(Diagnostics.Unexpected_0_Did_you_mean_to_escape_it_with_backslash, start, 1, String.fromCharCode(ch));
                                     isPreviousTermQuantifiable = true;
                                     break;
                                 }
                             }
-                            if (max && Number.parseInt(min) > Number.parseInt(max)) {
+                            else if (max && Number.parseInt(min) > Number.parseInt(max) && (!annexB || text.charCodeAt(pos) === CharacterCodes.closeBrace)) {
                                 error(Diagnostics.Numbers_out_of_order_in_quantifier, digitsStart, pos - digitsStart);
                             }
                         }
                         else if (!min) {
-                            if (unicodeMode) {
+                            if (!annexB) {

```

Comment:
Though it may be redundant, I think it might be better to still indicate `unicodeMode` here so that someone editing this code in the future doesn't mistakenly think this only applies to non-Annex B code. It may be better to use `unicodeMode || !annexB` and remove the `if (unicodeMode) { annexB = false; }` at the top of `scanRegularExpressionWorker`.

The same would go for other uses of `annexB` here.

Reply:
Are you sure you are really fine with a dozen of occurrences of `unicodeMode || !annexB`?

Reply:
No, but 
```ts
const anyUnicodeModeOrNonAnnexB = unicode Mode || !annexB;
```
would work.